### DISCUSSION TITLE
Social RSS reader

### DISCUSSION THREAD

**Discussion by erlend-sh** [muni-town/roomy#391] [Ideas]
Roomy/Weird loves [links](https://github.com/commune-os/weird/discussions/64).

As part of our [Read/Write premium tiers](https://github.com/commune-os/weird/discussions/14#discussioncomment-9221264), we want to integrate closely with an RSS reader since that’s a great way to feed links into a Weird profile.

We've already got a demo set up for https://miniflux.weird.one/ which can be logged into with ones Weird ID. However, since we want to make the rss-reader experience a core competency of our service, we’d like to move towards a completely ‘in-house’ solution for our default offering. 

To that end we are looking at Collie as a first step towards our own, rusty app. We’d love to coordinate directly with @parksb on this, but since he hasn’t responded to my messages yet we will start with a fork.

The essential flow we want to support entails the user logging into their local Collie app with their Weird ID in order to access their feed from any device.

TODO:
- https://github.com/parksb/collie/issues/14
- Basic user accounts in Collie
- [Rauthy client](https://crates.io/crates/rauthy-client) for login with Weird.

Later:
- View the publicly shared OPML feeds of your Weird friends, so as to find new stuff to follow for yourself, or suggest new things for your friends to follow. 
- Web app, I.e. collie.weird.one
- [Mobile app](https://github.com/tauri-apps/cargo-mobile2) (surely out of scope, but worth getting familiar with)
  - https://v2.tauri.app/start/prerequisites/#configure-for-mobile-targets

**Comment by parksb** [comment]
As you mentioned in Discord, Collie is a local-first app. I had assumed only a self-hosted sync server for https://github.com/parksb/collie/issues/14. However, this idea looks good to me. It seems different from Colie's original identity, so how about forking the repository?

**Reply by erlend-sh** [reply]
Yep, a fork is fine with us.

> I had assumed only a self-hosted sync server

that was my original idea as well, and if it’s self-contained enough maybe we can still upstream this part to Collie regular?

I’m just not sure what type of alternative system will be required for the user to bind their synced apps together, if not an account.

**Comment by zicklag** [comment]
Thinking about this: https://blog.erlend.sh/follow-anyone

That might be a good add-on, paid feature in Roomy. Since it naturally requires extra services to go and fetch things like RSS feeds.

At the same time you could also do something funny where whenever an admin opens Roomy it acts like a "local-first" kind of feed reader, and goes and fetches the RSS feeds on the client and then uses the admin's account to update the space's feed. 

For a personal space, where you are always the admin, it'd act like a local feed reader.

For community spaces the space would only get the feed updated when the admin gets on and the client auto-downloads the latest items from the feed.
